src/function/arithmetic/gcd.js

```diff
@@ -68,7 +65,30 @@ export const createGcd = /* #__PURE__ */ factory(name, dependencies, ({ typed, m
       DS: matAlgo01xDSid,
       Ss: matAlgo10xSids
     }),
-    gcdManySignature
+    {
+      [gcdManyTypesSignature]: typed.referToSelf(self => (a, b, args) => {
+        let res = self(a, b)
+        for (let i = 0; i < args.length; i++) {
+          res = self(res, args[i])
+        }
+        return res
+      }),
+      Array: typed.referToSelf(self => (args) => {
+        if (is1d(args)) {
+          return self(...args)
+        }
+        throw new ArgumentsError('gcd() supports only 1d matrices!')
+      }),
+      Matrix: typed.referToSelf(self => (args) => {
+        if (args.size().length === 1) {
+          return self(args._data)
+        }
+        if (args.size()[0] === 1) {
```

Comment:
This is a nice idea, supporting a 2d matrix in case it holds a vector. 

If we go for this though, I think we should work this out consistently: support both Array and Matrix, and support both `1xn` and `nx1` matrices, and maybe also support n-d matrices that hold a vector (i.e. all but one of it's dimensions holds a number > 1). What do you think?

Reply:
@josdejong I've a change with support for nested 1d arrays.

For `nx1` matrices, I guess you mean something like `[[1],[2],[3]]`. I think even though logically it represents a vector, the method should not support it. From my experiences with NumPy when doing matrix operations you sometimes may end up with a matrix that was accidentally rotated. Supporting only the form of `[1,2,3]` and `[[1,2,3]]` would help the user to keep the matrices always rotated properly

Reply:
Makes sense, sounds good to me. You can always transpose or flatten your matrix if you have a matrix like `[[1],[2],[3]]`.

Thanks for making the behavior consistent between Matrix and Array.